### PR TITLE
Async: convert buffer to str before pickling

### PR DESCRIPTION
**Pull request by sevein**
Connected to #358.

### CODE REVIEW

File: storage_service/locations/models/async.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 # stdlib, alphabetical
 import cPickle
```
Comment (jrwdunham) [inline storage_service/locations/models/async.py]: In Py3 `cPickle` is just `pickle` right, so if we're going to anticipate supporting Python 3 here, should we import `pickle` as `cPickle`?

File: storage_service/locations/models/async.py [hunk 2]
```diff
@@ -33,7 +34,14 @@ class Async(models.Model):
...
     @property
     def result(self):
-        return cPickle.loads(self._result)
+        result = self._result
+        # Convert memoryview (PY3) to str.
+        if sys.version_info[0] == 3 and isinstance(result, memoryview):
+            result = str(result)
+        # Convert buffer (PY2) to str.
+        if sys.version_info[0] == 2 and isinstance(result, buffer):
+            result = str(result)
```
Comment (jrwdunham) [inline storage_service/locations/models/async.py]: It looks like you could simplify using this by using `from django.utils import six`:

    if isinstance(result, six.memoryview):
        result = str(result)

**Review by jrwdunham — APPROVED** [review]
LGTM. One question and a suggestion for a possible improvement.

File: storage_service/locations/models/async.py (hunk 2 above)
Comment (jrwdunham) [reply storage_service/locations/models/async.py]: Cf. https://docs.djangoproject.com/en/2.0/_modules/django/utils/six/

**Comment by sevein** [comment]
@jrwdunham, ok done.

**Pull request merged**